File: seed/ruby-sdk-v2/simple-fhir/lib/fern_simple_fhir/types/account.rb
# frozen_string_literal: true

module FernSimpleFhir
  module Types
    class Account < Internal::Types::Model
      field :resource_type, -> { String }, optional: false, nullable: false
      field :name, -> { String }, optional: false, nullable: false
      field :patient, -> { FernSimpleFhir::Types::Patient }, optional: true, nullable: false
      field :practitioner, -> { FernSimpleFhir::Types::Practitioner }, optional: true, nullable: false
    end
  end
end

File: seed/pydantic-v2/simple-fhir/baseline/types/account.py
from pydantic import BaseModel
from .types.base_resource import BaseResource
from typing import Optional
from .types.patient import Patient
from .types.practitioner import Practitioner
from dt import datetime
from core.datetime_utils import serialize_datetime


class Account(BaseModel, BaseResource):
    resource_type: str
    name: str
    patient: Optional[Patient] = None
    practitioner: Optional[Practitioner] = None

    class Config:
        frozen = True
        smart_union = True
        json_encoders = {datetime: serialize_datetime}
